pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -160,9 +159,11 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 
 	for _, stage := range stages {
 		if err := stage(instance); err != nil {
+			r.exposeMetrics(instance)
 			return reconcile.Result{}, err
 		}
 	}
+	r.exposeMetrics(instance)
```

Comment:
Does it make more sense to call this once from `updateStatus`. Most of the stage functions -- the ones that may change the instance status -- defer a call to that anyway.

Reply:
Thank you
I have changed according to your comment